include/toolchain/gcc.h

```diff
@@ -55,16 +55,16 @@
 __extension__ ({							\
 	struct  __attribute__((__packed__)) {				\
 		__typeof__(*(p)) __v;					\
-	} *__p = (__typeof__(__p)) (p);					\
-	__p->__v;							\
+	} *__pg = (__typeof__(__pg)) (p);					\
```

Comment:
How does changing `__p` to `__pg` solve the issue?

Reply:
There is another macro using __p just after that, they are shadowing each other when used in the same context.

Reply:
That's curious, because the compound statement creates a new scope, and things declared in it shouldn't leak to the outer scope.   I would have to check the C standard to confirm this.

In any case, if it leaks, if one uses two `UNALIGNED_GET()` in succession, there will be two different `__pg`, one shadowing the other; this is specially bad if unaligned gets are performed with different data types.

Reply:
good point, let me check again what was happening there.